src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+        trip.thursday = true;
+        trip.friday = false;
+        trip.saturday = true;
+        trip.sunday = true;
+
+        List<String> checkedDates = ItineraryUtils.getDatesToCheckItineraryExistence(trip, testCase.checkAllDays);
+        Assertions.assertEquals(testCase.dates, checkedDates);
+    }
+
+    private static List<GetDatesTestCase> createGetDatesTestCases() {
+        // Each list includes dates to be monitored in a 7-day window starting from the query date.
+        return List.of(
+            // Dates solely based on monitored days (see the trip variable in the corresponding test).
+            new GetDatesTestCase(false, List.of(QUERY_DATE /* Thursday */, "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18")),
+
+            // If we forceAllDays to ItineraryUtils.getDatesToCheckItineraryExistence,
+            // it should return all dates in the 7-day window regardless of the ones set in the monitored trip.
+            new GetDatesTestCase(true, List.of(QUERY_DATE /* Thursday */, "2020-08-14", "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18", "2020-08-19"))
+        );
+    }
+
+    /**
+     * Check that the ignoreRealtime query parameter is set to true
+     * regardless of whether it was originally missing or false.
+     */
+    @Test
+    public void testAddIgnoreRealtimeParam() throws URISyntaxException {
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            MonitoredTrip trip = new MonitoredTrip();
+            trip.queryParams = query;
+            Map<String, String> params = ItineraryUtils.excludeRealtime(trip.parseQueryParams());
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    /**
+     * Check that a given trip is modified with an OTP itinerary that exists,
+     * right now based on the index of the selected itinerary in the UI.
+     */
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws URISyntaxException {
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = BASE_QUERY + "&ui_activeItinerary=1";
+        trip.itinerary = null;
+
+        List<Itinerary> itineraries = otpDispatcherPlanResponse.getResponse().plan.itineraries;
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createSameDayTestCases")
+    void testIsSameDay(SameDayTestCase testCase) throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(testCase.isArrival);
+        // The time zone for trip (and testCase.tripTime) is US Eastern per trip location.
+        ZoneId zoneId = trip.timezoneForTargetLocation();
```

Comment:
I removed this ZoneId lookup in #75 as it turns out that this is not how OTP deals with timezones. Instead you'll want to use `DateTimeUtils. getOtpZoneId`.